opentelemetry-exporter-cloud-trace/tests/test_gcp_resource_detector.py

```diff
@@ -46,11 +57,75 @@ def test_finding_gce_resources(self, getter):
         )
 
 
+def pop_environ_key(key):
+    if key in os.environ:
+        os.environ.pop(key)
+
+
+def clear_gke_env_vars():
+    pop_environ_key("CONTAINER_NAME")
+    pop_environ_key("NAMESPACE")
+    pop_environ_key("HOSTNAME")
+
+
+class TestGKEResourceFinder(unittest.TestCase):
+    def tearDown(self) -> None:
+        clear_gke_env_vars()
+
+    @mock.patch("opentelemetry.tools.resource_detector.requests.get")
+    def test_missing_environment_variables(self, getter):
+        getter.return_value.json.return_value = GKE_RESOURCES_JSON_STRING
```

Comment:
This line seems common to all the tests; is it possible to move it to a `setUp` method?

Reply:
I don't believe we can move it to setUp, but we can add the decorater to the class instead of each individual test function. (which I've done)